pkg/util/httputil/roundtrip.go

```diff
@@ -1,17 +1,20 @@
 package httputil
 
-import "net/http"
+import (
+	"net/http"
+)
 
 type userAgentTransport struct {
+	http.RoundTripper
+
 	userAgent string
-	base      http.RoundTripper
 }
 
 // NewUserAgentTransport creates a RoundTripper that attaches the configured user agent.
 func NewUserAgentTransport(userAgent string, base http.RoundTripper) http.RoundTripper {
 	return &userAgentTransport{
-		userAgent: userAgent,
-		base:      base,
+		RoundTripper: base,
+		userAgent:    userAgent,
```

Comment:
What's going on with this `base: base` to `RoundTripper: base` change?

Reply:
It makes the base the actual base. When I made this change originally, I was originally moving most of the construction inside this type, but given that this just overrides the roundtripper, it made more sense just to move the HTTPClient defaults back out to the provider. This is what happened when I put it back together, and the way that it existed when I tested.

Reply:
Semantically identical, I'd be glad to change it back if you want to roll the dice :)

Reply:
Huh. You're definitely the expert here, and I trust that expertise.

Reply:
No, you're right - I just sloppily threw it back together. The tests do pass, but I won't chance it. Will need to test again, but I'll do that before merging.